### PR TITLE
PanelEvents: Isolating angular panel events into it's own event bus + more event refactoring 

### LINKED ISSUE #29815: PanelEdit: After changing from angular to react panel the panel options are no longer updated on change
This is caused by the clean up of the event bus we need to do when removing the angular directive, we need to remove event subscriptions done by the directive. 

https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/state/PanelModel.ts#L485

### PR DESCRIPTION
**Pull request by torkelo**
Fixes #29815 

We have some problems with panel events, for angular we have to destroy and unsub to all the events subscribed by the angular controller and plugin code, but the PanelEditor will then lose its subscriptions. Feels like it's best to separate these into two different emitters. Then it becomes easy to clean up the angular state without any messy logic. 

This also updates the core Refresh & Render events to the new model and removes events that we no longer need.

### CODE REVIEW

File: public/app/features/dashboard/components/Inspector/QueryInspector.tsx [hunk 1]
```diff
@@ -39,9 +40,8 @@ interface State {
 }
 
 export class QueryInspector extends PureComponent<Props, State> {
-  formattedJson: any;
-  clipboard: any;
-  subscription?: Unsubscribable;
+  private formattedJson: any;
+  private subs = new Subscription();
```
Comment (hugohaggmark) [inline public/app/features/dashboard/components/Inspector/QueryInspector.tsx:44]: nit: shouldn't this initialization move into the constructor?

File: public/app/features/dashboard/dashgrid/DashboardGrid.tsx [hunk 2]
```diff
@@ -164,12 +161,6 @@ export class DashboardGrid extends PureComponent<Props> {
     this.forceUpdate();
   };
 
-  onWidthChange = () => {
-    for (const panel of this.props.dashboard.panels) {
-      panel.resizeDone();
-    }
-  };
-
```
Comment (hugohaggmark) [inline public/app/features/dashboard/dashgrid/DashboardGrid.tsx:172]: Out of curiosity, where is this done now?

File: public/app/features/dashboard/dashgrid/PanelChrome.tsx [hunk 3]
```diff
@@ -52,9 +52,8 @@ export interface State {
 }
 
 export class PanelChrome extends Component<Props, State> {
-  readonly timeSrv: TimeSrv = getTimeSrv();
-
-  querySubscription?: Unsubscribable;
+  private readonly timeSrv: TimeSrv = getTimeSrv();
+  private subs = new Subscription();
```
Comment (hugohaggmark) [inline public/app/features/dashboard/dashgrid/PanelChrome.tsx:56]: Same nit as above :)

File: public/app/features/dashboard/state/DashboardModel.ts [hunk 4]
```diff
@@ -883,10 +881,12 @@ export class DashboardModel {
     return rowPanels;
   }
 
+  /** @deprecated */
```
Comment (hugohaggmark) [inline public/app/features/dashboard/state/DashboardModel.ts:884]: Maybe we should also add a deprecation warning to these functions too?

File: public/app/features/panel/panel_directive.ts [hunk 5]
```diff
@@ -22,6 +24,7 @@ module.directive('grafanaPanel', ($rootScope, $document, $timeout) => {
       const panel: PanelModel = scope.ctrl.panel;
 
       let panelScrollbar: any;
+      let subs = new Subscription();
```
Comment (hugohaggmark) [inline public/app/features/panel/panel_directive.ts:27]: Why let?

**Review by hugohaggmark — APPROVED** [review]
Looks fantastic, so much cleaner now, great work! I did some testing, and I couldn't find any issues. I left some super-small nits for you to consider.

File: public/app/features/dashboard/dashgrid/DashboardGrid.tsx (hunk 2 above)
Comment (hugohaggmark) [reply public/app/features/dashboard/dashgrid/DashboardGrid.tsx:172]: Nevermind I saw later in the review where this is done now :)

File: public/app/features/dashboard/components/Inspector/QueryInspector.tsx (hunk 1 above)
Comment (torkelo) [reply public/app/features/dashboard/components/Inspector/QueryInspector.tsx:44]: yes, conflicted on what is best. This is technically the same as the constructor. The downside of moving it to the constructor is that I need to type it here and then duplicate it in the constructor. So this feels a bit leaner this way, sort of like declaring an instance value variable.

File: public/app/features/panel/panel_directive.ts (hunk 5 above)
Comment (torkelo) [reply public/app/features/panel/panel_directive.ts:27]: mistake, good catch!

File: public/app/features/dashboard/state/DashboardModel.ts (hunk 4 above)
Comment (torkelo) [reply public/app/features/dashboard/state/DashboardModel.ts:884]: good idea, added!

**Pull request merged**